implementations/rust/ockam/ockam_node/src/storage/database/sqlx_database.rs

```diff
@@ -114,54 +175,64 @@ impl SqlxDatabase {
         migration_set: impl MigrationSet,
     ) -> Result<Self> {
         debug!("create an in memory database for {usage}");
+        let configuration = DatabaseConfiguration::sqlite_in_memory();
         let pool = Self::create_in_memory_connection_pool().await?;
         let migrator = migration_set.create_migrator()?;
         migrator.migrate(&pool).await?;
         // FIXME: We should be careful if we run multiple nodes in one process
         let db = SqlxDatabase {
             pool: Arc::new(pool),
-            path: None,
+            configuration,
         };
         Ok(db)
     }
 
-    async fn create_at(path: &Path) -> Result<Self> {
-        let path = path.to_path_buf();
+    async fn create_at(configuration: DatabaseConfiguration) -> Result<Self> {
         // Creates database file if it doesn't exist
-        let pool = Self::create_connection_pool(path.as_path()).await?;
+        let pool = Self::create_connection_pool(configuration.clone()).await?;
         Ok(SqlxDatabase {
             pool: Arc::new(pool),
-            path: Some(path),
+            configuration,
         })
     }
 
-    pub(crate) async fn create_connection_pool(path: &Path) -> Result<SqlitePool> {
-        let options = SqliteConnectOptions::new()
-            .filename(path)
-            .create_if_missing(true)
-            .log_statements(LevelFilter::Trace)
-            .log_slow_statements(LevelFilter::Trace, Duration::from_secs(1));
-        let pool = SqlitePool::connect_with(options)
+    pub(crate) async fn create_connection_pool(
+        configuration: DatabaseConfiguration,
```

Comment:
Can we use a reference instead of the value here?

Reply:
Done.